stackstate_checks_base/stackstate_checks/base/checks/base.py

```diff
@@ -482,6 +484,56 @@ def relation(self, source, target, type, data, streams=None, checks=None):
         data = self._map_relation_data(source, target, type, fixed_data, fixed_streams, fixed_checks)
         topology.submit_relation(self, self.check_id, self._get_instance_key_dict(), source, target, type, data)
 
+    def _map_stackstate_tags_and_instance_config(self, data):
+        # Get the first instance and create a deep copy
+        instance = self.instances[0] if self.instances is not None and len(self.instances) else {}
+        check_instance = copy.deepcopy(instance)
+
+        # Extract or create the tags and identifier objects
+        tags = data.get('tags', [])
+        identifiers = data.get("identifiers", [])
+
+        # Find the stackstate-identifiers within tags
+        identifier_tag = next((tag for tag in tags if ("stackstate-identifiers:" in tag)), None)
+
+        # We attempt to split and map out the identifiers specified in the identifier_tag
+        # ** Does not support config **
+        if isinstance(identifier_tag, str):
+            """
+            We are testing the following with the regex block below
+            - Match all spaces if any at the start followed by a must have comma then followed again by a possible
+                single or multiple spaces
+            - Second possibility is a single or multiple space only without a comma
+            The resulting regex will work on a complex line like the following example:
+                Input: a, b, c,d,e f g h, i , j ,k   ,l  ,  m
+                Result: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm']
+            """
+            identifier_tag_content = identifier_tag.split('stackstate-identifiers:')[1]
+            if len(identifier_tag_content) > 0:
+                identifier_list = re.split('(?:\\s+)?,(?:\\s+)?|\\s+', identifier_tag_content)
+                clean_identifier_list = list(filter(lambda item: len(item) > 0, identifier_list))
+                data["identifiers"] = identifiers + clean_identifier_list
+
+        # Generic mapping function for tags or config
+        # Attempt to find if the target exists on a tag or config and map that to the origin value
+        # There's a optional default value if required
+        # Value override order: tags < config.yaml
+        def _map_config_or_tag(target, origin, is_array=False, default=None):
```

Comment:
Is there a way to test all parameters combinations? I see there is `default` parameter that is not used in any function call.

Reply:
Moved the function out of the nesting and added unit testing for all possible options.